### ISSUE TITLE
Running in the background on Android

### ISSUE THREAD

**Issue by easyaspi314** [luanti-org/luanti#13115]
## Problem

When pressing the home button on Android, the game seemingly continues to run in the background. There is no sticky notification to indicate this, and I was entirely unaware until I had the mineclone2 world select menu open and heard music in the background. This can cause battery drain.

## Solutions

This should probably be configurable.

On singleplayer worlds and menus, if the user presses the home button the game should probably be paused and frozen, and on multiplayer worlds it should probably disconnect.

## Alternatives

Alternatively, Minetest should have a sticky notification that indicates it is running.

**Comment by nephele-gh** [comment]
A sticky notification on android is sort of an anti-pattern applications employ, this stops android from specific sorts of battery optimizations. As it makes the UI much clunkier I think this shouldn't be added at all.
The application overview already makes it clear that the application is still running and it's state

**Comment by sfan5** [comment]
I tend to agree, however if Android kills the game while it's in the background it won't be shut down properly and restarting it is far from seamless.
Does Minetest already enter the pause dialog when backgrounded?

**Comment by Unacceptium** [comment]
i like that mt stays opened, because i can check websites, or get a music to play while mt is running in background. but closing it isn't that hard, you just swipe it from the recent apps. for me, this is better than ok, but a notification indicating that minetest is still running, just in the background, would be a not that bad idea, especially for less experienced android users. if android kills the game in the background, you just need to disable battery optimization in the system's app settings.(works for me 100% even on android 12)

**Comment by srifqi** [comment]
Minetest for Android was fully paused (as in not responding) when the main game app is unfocused (see Issue #10842). After PR minetest/irrlicht#255, that is no longer the case. PR #14058 attempts to pause rendering while still running networking code and other non-rendering code (e.g. server code for singleplayer/as a host).

**Comment by Unacceptium** [comment]
@srifqi i don't even understand a word you said but i'm totally ok with the game going "unfocused" and just lowers the fps.
like i can hear when someone hit me of an advtrain passes, even if i'm in a different app.

**Comment by srifqi** [comment]
Ah, sorry for that. It was midnight here. 💤

By the way, since the 5.8.0 is already out, have you tested it? What changes do we want/need to have regarding this issue?

**Comment by grorp** [comment]
>  like i can hear when someone hit me of an advtrain passes, even if i'm in a different app.

This should still work after #14058, sounds are not affected by that PR. However, the sounds will be quieter after #14083 :)

> By the way, since the 5.8.0 is already out, have you tested it? What changes do we want/need to have regarding this issue?

From Unacceptium's feedback, it sounds to me like the current behavior is already what they want (the skipped rendering should, after all, not be observable), except for this part:

> but a notification indicating that minetest is still running, just in the background, would be a not that bad idea, especially for less experienced android users

I'm honestly not sure about #13125, but I haven't yet tried it or thought about it much. What especially bothers me is this:

> Known issue: The notification still exists after removing Minetest from the recent application list while the player is still inside a world/server. Restarting Minetest (or via tapping the notification) will clear it up. This issue also exists in other application and seems to be OEM-dependent.

No notification might be better than a buggy notification.

**Comment by Unacceptium** [comment]
@grorp wrote:
"From Unacceptium's feedback, it sounds to me like the current behavior is already what they want (the skipped rendering should, after all, not be observable), except for this part:"

1-they? i Directly stated that this is only my opinion!
2- i hate that your fork full stops in background
3- is these features are in a specific version of MT? because i don't really follow these updates, the official MT is unplayable for me (jump button placement)

"No notification might be better than a buggy notification."
more experienced android coders did worked around that OEM issue and for example, VLC media player can run in background, full responging to notification buttons, and SWIPED from the Recent apps tab.
and a better example, "ninja" browser. it is not in recent app, no notification, no battey optimization disabled, only background running enabled and the app stays open with every link you opened. in background.

also, the user Should disable battery optimization and allow MT to run in background. these can be found when user long press the app icon. thats how i play and mt is fine for me.

Again, this Only is my opinion.

**Comment by Unacceptium** [comment]
@srifqi 
> Ah, sorry for that. It was midnight here. 💤
> 
> By the way, since the 5.8.0 is already out, have you tested it? What changes do we want/need to have regarding this issue?

i installed 5.8:
the game never get unfocused,
it runs in background with Every frame processed
i sent it to background for a min and this happened:
i hear trains passing. and getting back to MT i am at the main menu. NO indications that my MT shut down, i heard the trains passing before i ooened MT!!!!!
for me this is bad.

MT does not get FPS unfocused at all. if notif.bar open, chat open, sent to background, etc. is this the goal?
but MT just shut down! from a sec to another with no indications!
for checking similarity, rollertest goes unfocused for all of these and if it shuts down before i get back, RT shows "shutting down" for a short second.

sorry for spamming this message.

**Comment by Unacceptium** [comment]
> > 1-they? i Directly stated that this is only my opinion!
> 
> the word "they" in this case is third person singular, not plural.

thats why i hate english
